packages/search/tests/unit/services/navi-search-provider-test.js

```diff
@@ -0,0 +1,48 @@
+import Service from '@ember/service';
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import setupMirage from 'ember-cli-mirage/test-support/setup-mirage';
+
+module('Unit | Service | navi-search-provider', function(hooks) {
+  setupTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function() {
+    // Load metadata needed for request fragment
+    await this.owner.lookup('service:bard-metadata').loadMetadata();
+    this.service = this.owner.lookup('service:navi-search-provider');
+    const store = this.owner.lookup('service:store'),
+      mockAuthor = store.createRecord('user', { id: 'ciela' });
+    this.owner.register(
+      'service:user',
+      Service.extend({
+        getUser: () => mockAuthor
+      })
+    );
+  });
+
+  test('it exists', function(assert) {
+    let service = this.owner.lookup('service:navi-search-provider');
+    assert.ok(service);
+  });
+
+  test('get all search providers', function(assert) {
+    let availableSearchProviders = this.service.all();
+    assert.equal(availableSearchProviders.length, 1, 'Discovered 1 search provider.');
```

Comment:
Can we check the properties of the returned search providers rather than just testing the length? `availableSearchProviders` could be `['foo']` and it would satisfy this test case.